src/storage/src/storage3/_async/bucket.py

```diff
@@ -3,6 +3,8 @@
 from typing import Any, Optional
 
 from httpx import AsyncClient, HTTPStatusError, Response
+from httpx._types import HeaderTypes
```

Comment:
I'm not keen on a import from private module here. But it's not a deal breaker.

Reply:
This is to support both `dict[str, str]` and `Headers` types.